src/lib/pages/backoffice/Document/DocumentForms/DocumentEditor/DocumentForm/components/MetadataExtensions.js

```diff
@@ -25,41 +27,98 @@ const getFormComponent = fieldType => {
   }
 };
 
+const sortByLine = (a, b) => {
+  if (a.line < b.line) {
+    return -1;
+  }
+  if (a.line > b.line) {
+    return 1;
+  }
+  return 0;
+};
+
 export const MetadataExtensions = ({ extensions }) => {
   const { label, fields } = invenioConfig.DOCUMENTS.extensions;
   const configDefaults = {};
   _keys(fields).map(key => (configDefaults[key] = fields[key]['default']));
   const allExtensions = _merge(configDefaults, extensions);
 
   let components = [];
+  let orderedComponents = [];
+  let lineComponents = [];
+  let oldLine = 1;
+
+  //Create an array with the components to display
   _forOwn(allExtensions, (value, key) => {
     const componentType = invenioConfig.DOCUMENTS.extensions.fields[key].type;
-    const Component = getFormComponent(componentType);
-
-    components.push(
-      <Component
-        fieldPath={`extensions.${key}`}
-        key={key}
-        label={_get(
-          invenioConfig,
-          `extensions.document.fields.${key}.label`,
-          key
-        )}
-        optimized
-        required={_get(
-          invenioConfig,
-          `extensions.document.fields.${key}.isRequired`,
-          false
-        )}
-      />
-    );
+    const line = invenioConfig.DOCUMENTS.extensions.fields[key].line;
+    const label = invenioConfig.DOCUMENTS.extensions.fields[key].label;
+    if (componentType === 'vocabulary') {
+      components.push({
+        component: (
+          <VocabularyField
+            type={invenioConfig.DOCUMENTS.extensions.fields[key].vocabularyType}
+            fieldPath={`extensions.${key}`}
+            label={label}
+            multiple
+            placeholder={`Select ${label} ...`}
+          />
+        ),
+        line: line,
+      });
+    } else {
+      const Component = getFormComponent(componentType);
+      components.push({
+        component: (
+          <Component
+            fieldPath={`extensions.${key}`}
+            key={key}
+            label={label}
+            optimized
+            toggle={componentType === 'boolean' ? true : undefined}
+            required={_get(
+              invenioConfig,
+              `extensions.document.fields.${key}.isRequired`,
+              false
+            )}
+          />
+        ),
+        line: line,
+      });
+    }
+  });
+  //Sort the components by line and group them
+  components.sort(sortByLine);
+  components.forEach(element => {
+    if (element.line === oldLine) {
+      lineComponents.push(element.component);
+    } else {
+      if (lineComponents) {
+        orderedComponents.push(
+          <GroupField widths="equal">
+            {lineComponents.map(element => element)}
```

Comment:
What happens to the ui if I have too many elements (not lines :))?

Reply:
All the fields we add we should add them with the line, if we did not add a line they will be displayed all in the same line (we can change this behaviour to a default layout, but I think the line attribute should be "mandatory")